src/main/java/community/solace/spring/integration/leader/leader/SolaceLeaderInitiator.java

```diff
@@ -222,21 +223,24 @@ private class LeaderGroupContainer {
         private LeaderGroupContainer(Candidate candidate, boolean yieldOnShutdown) {
             this.candidate = candidate;
 
+            // Work around for a strange mokito bug. Mockito returns null when called more than 10 tims.

```

Comment:
Lets have a look at this together, if prod code has to be filled with workarounds for tests it smells :-)